tree/ntuple/v7/inc/ROOT/RNTupleOptions.hxx

```diff
@@ -61,6 +61,9 @@ protected:
    /// fApproxUnzippedPageSize/2 and fApproxUnzippedPageSize * 1.5 in size.
    std::size_t fApproxUnzippedPageSize = 64 * 1024;
    bool fUseBufferedWrite = true;
+   /// If set, 64bit index columns are replaced by 32bit index columns. This limits the cluster size to 512MB
```

Comment:
Why 512MB?  A 32bit unsigned integer should be able to encode an offset in the range $[0, 4GB]$, no?

Reply:
This is due to bit granularity of columns. In an std::vector<bool>, a 32bit index column can address a bit array of `bool`s up to 512M.

Reply:
I am still not quite following.  What do you mean by 'granularity' and how does it affect the index?

Reply:
What I meant is that in a cluster of 512MB, we can store 2^32 booleans, provided the cluster contains a single bool column.

Reply:
I see [it is, of course, because we can store 8 booleans in 1B]